pkg/flow/internal/controller/node_declare.go

```diff
@@ -0,0 +1,51 @@
+package controller
+
+import (
+	"sync"
+
+	"github.com/grafana/river/ast"
+	"github.com/grafana/river/vm"
+)
+
+type DeclareNode struct {
+	label         string
+	nodeID        string
+	componentName string
+	declare       *Declare
+	mut           sync.RWMutex
```

Comment:
`mut` should be grouped with the fields that it guards. For example, if `mut` is guarding `declare` then:

```
type DeclareNode struct {
	label         string
	nodeID        string
	componentName string

	mut           sync.RWMutex
	declare       *Declare
}
```